### ISSUE TITLE
`waitForIdle` does not wait for all queries

### ISSUE THREAD

**Issue by raffomania** [Prior99/mobx-repository#25]
`waitForIdle` currently works by waiting for promises in `listenersById` and `listenersByQuery`. These promises are created in `waitForId` and `waitForQuery`, respectively.
However, in `loadById` and `loadByQuery`, `waitForQuery` is only called for requests with a `RequestStatus` of `IN_PROGRESS` or `ERROR`. For requests that are dispatched for the first time, `waitForQuery` is not called, no listeners are registered, and `waitForIdle` resolves while these requests are still in progress.

Would you be interested in extending `waitForIdle` or adding a second method allowing users to wait for all requests? 
Off the top of my head, I have two naive ideas on how that could be implemented:

- Call `waitForId`/`waitForQuery` for every entry in `stateById`/`stateByQuery` with a status of `IN_PROGRESS` and await those calls in `waitForIdle`
- Iterating over every entry in `stateById`/`stateByQuery`, use `when` from mobx to resolve a promise once no entries with a status of `IN_PROGRESS` are left

If you like, I can open a PR for one of these.

**Comment by Prior99** [comment]
Yes, I would very much welcome a pull request fixing this bug. I think the first approach sounds good.
